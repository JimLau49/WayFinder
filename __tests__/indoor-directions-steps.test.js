import React from "react";

import IndoorDirectionSteps from "../src/components/travel-route/indoor-directions-steps.component";
import PathFindingService from "../src/services/pathfinding.service";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import * as testData from "./__testData__/indoor-travel-route";

const mockFloorPath = PathFindingService.getInstance().findPathBetweenPOIs(
  testData.mockPOIs[0],
  testData.mockPOIs[1],
  false
);

describe("Indoor Directions Text component", () => {
  it("should match indoor travel directions steps snapshot", () => {
    const mockOnCloseTravelSteps = jest.fn();
    const tree = renderer
      .create(
        <IndoorDirectionSteps
          floorPaths={mockFloorPath}
          onCloseTravelSteps={mockOnCloseTravelSteps}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should not render if the floor plans are null", () => {
    const tree = renderer
      .create(
        <IndoorDirectionSteps floorPaths={null} onCloseTravelSteps={null} />
      )
      .toJSON();
    expect(tree).toBe(null);
  });

  it("should call mockOnCloseTravelSteps on 'x' button press", () => {
    const mockOnCloseTravelSteps = jest.fn();
    const wrapper = shallow(
      <IndoorDirectionSteps
        floorPaths={mockFloorPath}
        onCloseTravelSteps={mockOnCloseTravelSteps}
      />
    );
    const button = wrapper.find("TouchableOpacity");
    // NOTE: Cannot currently test
    // Waiting for answer: https://github.com/octopitus/rn-sliding-up-panel/issues/165
    // button.simulate("press");
  });
});
