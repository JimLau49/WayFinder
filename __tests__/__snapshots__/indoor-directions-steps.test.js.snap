// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indoor Directions Text component should match snapshot 1`] = `
<View
  disableScrollViewPanResponder={true}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "alignSelf": "stretch",
      "bottom": -1292,
      "flex": 1,
      "height": 1292,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "transform": Array [
        Object {
          "translateY": -113,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#AA2B45",
          "display": "flex",
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
      testID="dirPanelCloseButton"
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 667,
          "justifyContent": "center",
        }
      }
      testID="dirPanel"
    >
      <View
        style={
          Object {
            "marginBottom": -7,
          }
        }
      >
        <Text />
      </View>
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "textAlign": "center",
          }
        }
        testID="dirPanelDisplayName"
      >
        Indoor Directions
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "width": "100%",
          }
        }
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        testID="dirScrollView"
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginTop": 10,
              }
            }
            testID="stepNumber0"
          >
            1
            . 
            Head to your Destination.
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
